src/sedpack/io/dataset.py

```diff
@@ -938,16 +959,20 @@ def as_numpy_iterator(  # pylint: disable=too-many-arguments
                                  f"{self.dataset_structure.shard_file_type}")
 
         example_iterator = itertools.chain.from_iterable(
-            map(shard_iterator.iterate_shard, shard_paths_iterator))
+            map(
+                shard_iterator.iterate_shard,  # type: ignore
+                shard_paths_iterator))  # type: ignore
 
         # Process each record if requested
         if process_record:
-            example_iterator = map(process_record, example_iterator)
+            example_iterator = map(process_record,
+                                   example_iterator)  # type: ignore
 
         # Randomize only if > 0 -- no shuffle in test/validation
         if shuffle:
-            example_iterator = shuffle_buffer(example_iterator,
-                                              buffer_size=shuffle)
+            example_iterator = shuffle_buffer(
+                example_iterator,  # type: ignore
```

Comment:
That's quite a lot of `type: ignore` lines in our code (i.e. not counting the `lz4` and other 3rd party library typing issues) at the end. Maybe have a bug to try to reduce their amount later?

Reply:
Issue #13

Reply:
Issue #14